gis/bylaw_text_to_centreline/README.md

```diff
@@ -0,0 +1,300 @@
+        )
+    
+    END )
+
+--routed centreline - part of centreline xx metres from that intersection that got trimmed
+WHEN ST_Within(ST_Transform(ST_ClosestPoint(_wip2.whole_centreline, _wip2.oid1_geom_translated), 2952), ST_BUFFER(ST_Transform(_wip2.pgrout_centreline, 2952), 2, 'endcap=square join=round')) = TRUE
+THEN (
+    CASE WHEN ST_LineLocatePoint(_wip2.whole_centreline, _wip2.oid1_geom)
+    < ST_LineLocatePoint(_wip2.whole_centreline, _wip2.oid1_geom_translated)
+    THEN ST_Difference(_wip2.pgrout_centreline ,
+    ST_LineSubstring(_wip2.pgrout_centreline, 
+    ST_LineLocatePoint(_wip2.pgrout_centreline, _wip2.oid1_geom),
+    ST_LineLocatePoint(_wip2.pgrout_centreline, _wip2.oid1_geom) + (metres_btwn1/ST_Length(ST_Transform(_wip2.pgrout_centreline, 2952)))  )
+        )
+
+    WHEN ST_LineLocatePoint(_wip2.whole_centreline, _wip2.oid1_geom)
+    > ST_LineLocatePoint(_wip2.whole_centreline, _wip2.oid1_geom_translated)
+    THEN ST_Difference(_wip2.pgrout_centreline ,
+    ST_LineSubstring(_wip2.pgrout_centreline, 
+    ST_LineLocatePoint(_wip2.pgrout_centreline, _wip2.oid1_geom) - (metres_btwn1/ST_Length(ST_Transform(_wip2.pgrout_centreline, 2952))),
+    ST_LineLocatePoint(_wip2.pgrout_centreline, _wip2.oid1_geom) )  
+        )
+
+    END )
+
+ELSE _wip2.pgrout_centreline
+
+END);
+```
+
+The image below shows the four possible cases (A1, A2, B1 and B2)
+![image](https://user-images.githubusercontent.com/54872846/77707649-284df380-6f9c-11ea-9509-b0db0cdf24c5.png)
+
+
+## Confidence output
+
+The function outputs a confidence level (`con`), which is defined after the `centreline_segments` variable. This value represents how close the names of the input values are to the intersections they were matched to. Specifically, this value is the sum of the levenshtien distance between the `highway2` value and the street name it matched to and the `btwn1`/`btwn2` value and the street name it matched to. It is the sum of this value for both intersections.
+
+# Quality Control
+
+There is a [Github Issue](https://github.com/CityofToronto/bdit_data-sources/issues/188) on weird bylaws that haven't been matched from Chelsea's work. \
+For example: \
+i) The streets stated in bylaws are slightly off i.e. they do not intersect or are rather far apart. \
+ii) Directionality matters. Northbound and Southbound route between two intersection are of different centrelines such as the Allen Rd. \
+iii) The error message `line_locate_point: 1st arg isn't a line` returned for 49 distinct id's. This mostly occurs when trying to slice a multilinestring. 
+
+Majority of them are also some issues that I have encountered with my new processes but have not been solved yet. I will list down the issue below and some idea on how we can tackle them.
+
+## pgRouting returns the shortest path but street name different from `highway`
```

Comment:
This was solved right?

Reply:
Kinda? Based on this [comment](https://github.com/CityofToronto/bdit_data-sources/issues/268#issuecomment-620352090), previously there were 84 bylaws which fell under this category, and I solved 69 of them but there are still 15 of them that still fails due to this reason. Am already using 0.3*cost for the pgRouting process

Reply:
This section should be updated with where that issue ended: only 15 of previously 84 bylaws fail there because the cost was modified to prefer the highway matching the streetname